cs/src/comm/epoxy-transport/EpoxyListener.cs

```diff
@@ -105,57 +106,69 @@ internal void InvokeOnDisconnected(DisconnectedEventArgs args)
             logger.Site().Information("Accepting connections on {0}", ListenEndpoint);
             while (!t.IsCancellationRequested)
             {
-                Socket socket = null;
-                EpoxyNetworkStream epoxyStream = null;
-
                 try
                 {
-                    socket = await listener.AcceptSocketAsync();
-                    logger.Site().Debug("Accepted connection from {0}.", socket.RemoteEndPoint);
-
-                    EpoxyTransport.ConfigureSocketKeepAlive(socket, timeoutConfig, logger);
-
-                    epoxyStream = await EpoxyNetworkStream.MakeServerStreamAsync(socket, tlsConfig, logger);
-                    socket = null; // epoxyStream now owns the socket
-
-                    var connection = EpoxyConnection.MakeServerConnection(
-                        parentTransport,
-                        this,
-                        serviceHost,
-                        epoxyStream,
-                        logger,
-                        metrics);
-
-                    // connection now owns the EpoxyNetworkStream
-                    epoxyStream = null;
-
-                    lock (connectionsLock)
-                    {
-                        connections.Add(connection);
-                    }
-
-                    await connection.StartAsync();
-                    logger.Site().Debug("Started server-side connection for {0}", connection.RemoteEndPoint);
-                }
-                catch (AuthenticationException ex)
-                {
-                    logger.Site().Error(ex, "Failed to authenticate remote connection from {0}", socket?.RemoteEndPoint);
-                    ShutdownSocketSafe(socket, epoxyStream);
+                    Socket socket = await listener.AcceptSocketAsync();
+                    EndPoint remoteEndpoint = socket.RemoteEndPoint;
+                    logger.Site().Debug("Accepted connection from {0}.", remoteEndpoint);
+
+                    // Don't need to wait for the connection to get fully established before
+                    // accepting the next one, so fire and forget (with logging) StartAsync.
+                    StartClientConnectionAsync(socket)
+                        .ContinueWith(startTask => LogClientStartResult(remoteEndpoint, startTask))
+                        .Forget();
```

Comment:
`var _ = Start()` doesn't work, because then we get a warning about an unused variable. I've inserted a `#pragma` to disable the unobserved task warning for this block of code instead.